code/controllers/subsystem/radiation.dm

```diff
@@ -16,3 +21,46 @@ PROCESSING_SUBSYSTEM_DEF(radiation)
 	SSblackbox.record_feedback("tally", "contaminated", 1, master.type)
 	var/msg = "has become contaminated with enough radiation to contaminate other objects. || Source: [contamination.source] || Strength: [contamination.strength]"
 	master.investigate_log(msg, "radiation")
+
+/datum/controller/subsystem/processing/radiation/proc/get_turf_radiation(turf/place)
+	return turf_cache.get_turf_radiation(place)
+
+/datum/controller/subsystem/processing/radiation/proc/update_rad_cache(datum/component/radioactive/thing)
+	return turf_cache.update_rad_cache(thing)
+/datum/radiation_cache
+	// Cache radiation levels for each turf so it doesn't need to be done iteratively
+	// turf_rad_cache is the state in the current loop, and may not be 100% representative
+	// until processing is complete.
+	// prev_rad_cache is the fully evaluated radiation state cache from the previous tick.
+	var/has_refreshed_cache = FALSE
+	var/list/turf_rad_cache = list()
+	var/list/prev_rad_cache = list()
```

Comment:
Yeah. `has_refreshed_cache` is vestigial and can be removed.

The reason I made a new datum was so I could have the lists be updated in the radiation processing loop, since the cur-tick cache needs to be dumped into the prev rad cache and cleared every time radioactive objects update. I was thinking of adding something to `fire()` but it was awkward to get something that'd only fire once per processing loop. I similarly figured it'd be cleaner to have the lists contained with the logic that's used to update them.

Reply:
Removed the extra var in 14cde58

Reply:
> I was thinking of adding something to fire() but it was awkward to get something that'd only fire once per processing loop

I'm not really sure what you mean here. You can put just do:
```dm
/datum/controller/subsystem/processing/radiation/fire(resumed)
	. = ..()
	refresh_rad_cache()
```
Which is basically identical to doing:
```dm
/datum/radiation_cache/process()
	refresh_rad_cache()
```
And the old `radiation_cache` procs can be converted (already mentioned that though). With the way it's set up right now, there's this sort of middle man that doesn't need to be there. Unless I'm horribly blind and missing something.. the `radiation_cache` datum can instead stuck onto SSradiation.

Reply:
In my testing with the debugger, `fire()` would be called once for every item in the processing queue. I tried doing that before, but it just meant that the rad cache would be cleared *on every item,* not just every tick of the radiation queue.

Reply:
That.. should not be the case. The `fire()` proc is called only once every `wait` period.

Reply:
That's what I saw when testing it before, but with a second look you're totally right. I'll see about refactoring out the datum.